typescript/testsuite/testsuite_impl/advanced_network_test/advanced_network_test_.ts

```diff
@@ -0,0 +1,75 @@
+import { TestNetwork } from "../../network_impl/test_network"
+import { TestConfigurationBuilder } from "kurtosis-testsuite-api-lib"; //TODO (Ali)
+import * as log from "loglevel";
+import { ok, err, Result } from "neverthrow";
+import { APIClient, Person } from "../../../api/api_service_client/api_client";
+
+const TEST_PERSON_ID: number = 46;
+
+export class AdvancedNetworkTest {
+	private readonly datastoreServiceImage: string;
+	private readonly apiServiceImage: string;
+	
+	constructor (datastoreServiceImage: string, apiServiceImage: string) {
+		this.datastoreServiceImage = datastoreServiceImage; 
+		this.apiServiceImage = apiServiceImage;
+	}
+
+	public configure(builder: TestConfigurationBuilder): void {
+		builder.withSetupTimeoutSeconds(60).withRunTimeoutSeconds(60); //TODO (Ali) - allowed since typescript gives direct reference
+	}
+
+	public async setup(networkCtx: NetworkContext): Promise<Result<Network, Error>> { //TODO (Ali) - allowed to be async
+		const network: TestNetwork = new TestNetwork(networkCtx, this.datastoreServiceImage, this.apiServiceImage);
+		// Note how setup logic has been pushed into a custom Network implementation, to make test-writing easy
+		
+		const setupDatastoreAndTwoApisResult: Result<null, Error> = await network.setupDatastoreAndTwoApis();
+		if (!setupDatastoreAndTwoApisResult.isOk()) {
+			return err(setupDatastoreAndTwoApisResult.error);
+		}
+		return ok(network);
+	}
+
+	public async run(network: Network): Promise<Result<null, Error>> { //TODO (Ali) - allowed to be async
+		const castedNetwork: TestNetwork = <TestNetwork>network;
+		const personModifierClientResult: Result<APIClient, Error> = castedNetwork.getPersonModifyingApiClient()
+		if (!personModifierClientResult.isOk()) {
+			return err(personModifierClientResult.error);
+		}
+		const personModifierClient: APIClient = personModifierClientResult.value;
+
+		const personRetrieverClientResult: Result<APIClient, Error> = castedNetwork.getPersonRetrievingApiClient()
+		if (!personRetrieverClientResult.isOk()) {
+			return err(personRetrieverClientResult.error);
+		}
+		const personRetrieverClient: APIClient = personRetrieverClientResult.value;
+
+		log.info("Adding test person via person-modifying API client...");
+		const addPersonResult: Result<null, Error> = await personModifierClient.addPerson(TEST_PERSON_ID);
+		if (!addPersonResult.isOk()) {
+			return err(addPersonResult.error);
+		}
+		log.info("Test person added");
+
+		log.info("Incrementing test person's number of books read through person-modifying API client...");
+		const incrementBooksReadResult: Result<null, Error> = await personModifierClient.incrementBooksRead(TEST_PERSON_ID);
+		if (!incrementBooksReadResult.isOk()) {
+			return err(incrementBooksReadResult.error);
+		}
+		log.info("Incremented number of books read");
+
+		log.info("Retrieving test person to verify number of books read person-retrieving API client...");
```

Comment:
Is this language correct? Seems like we're missing something between `books read` and `person-retrieving API client`... if the Go version is goofed-up too, we should fix both

Reply:
Ah yep, in both. I will update this to `"Retrieving test person to verify number of books read by the person-retrieving API client..."`

![Screen Shot 2021-08-25 at 10 02 53 AM](https://user-images.githubusercontent.com/52847195/130805101-d2dd1f9b-d2c2-4196-a75b-b4e987d188d1.png)